test/mocks/data/challenge_data.dart

```diff
@@ -0,0 +1,21 @@
+import "package:cloud_firestore/cloud_firestore.dart";
+import "package:proxima/models/database/challenge/challenge_data.dart";
+
+class ChallengeGenerator {
+  ChallengeGenerator._();
+
+  /// Generates a challenge
+  /// by default, the challenge is not completed and expires now
+  /// [isCompleted] whether the challenge is completed
+  /// [expirationDelay] the delay until the challenge expires. The challenge will expire at DateTime.now() + expirationDelay
```

Comment:
This does not meet Dart's "DO use prose to explain parameters, return values, and exceptions" [documentation convention](https://dart.dev/effective-dart/documentation#do-use-prose-to-explain-parameters-return-values-and-exceptions). Here is a potential fix:
```suggestion
  /// Generates a challenge (using mock data).
  /// It expires in [expirationDelay] (i.e., it expires at `DateTime.now() + expirationDelay`), 
  /// and the generated challenge may be completed or not, depending on [isCompleted].
```